source/pidomus.cc

```diff
@@ -523,6 +523,68 @@ void piDoMUS<dim, spacedim, LAC>::assemble_matrices (const double t,
 /* ------------------------ MESH AND GRID ------------------------ */
 
 template <int dim, int spacedim, typename LAC>
+void piDoMUS<dim, spacedim, LAC>::refine_and_transfer_solutions(LATrilinos::VectorType &y,
+    LATrilinos::VectorType &y_dot,
+    LATrilinos::VectorType &distributed_y,
+    LATrilinos::VectorType &distributed_y_dot,
+    bool adaptive_refinement)
+{
+  distributed_y = y;
+  distributed_y_dot = y_dot;
+  parallel::distributed::SolutionTransfer<dim, LATrilinos::VectorType, DoFHandler<dim,spacedim> > sol_tr(*dof_handler);
+  parallel::distributed::SolutionTransfer<dim, LATrilinos::VectorType, DoFHandler<dim,spacedim> > sol_dot_tr(*dof_handler);
+  LATrilinos::VectorType sol (distributed_y);
+  LATrilinos::VectorType sol_dot (distributed_y_dot);
+  // sol = y;
+  // sol_dot = y_dot;
+
+  triangulation->prepare_coarsening_and_refinement();
+  sol_tr.prepare_for_coarsening_and_refinement (sol);
+  sol_dot_tr.prepare_for_coarsening_and_refinement(sol_dot);
+  if (adaptive_refinement)
+    triangulation->execute_coarsening_and_refinement ();
+  else
+    triangulation->refine_global (1);
+
+  setup_dofs(false);
+
+  LATrilinos::VectorType tmp (y);
+  LATrilinos::VectorType tmp_dot (y_dot);
+
+  sol_tr.interpolate (tmp);
+  sol_dot_tr.interpolate (tmp_dot);
+  y = tmp;
+  y_dot = tmp_dot;
+}
+
+template <int dim, int spacedim, typename LAC>
+void piDoMUS<dim, spacedim, LAC>::refine_and_transfer_solutions(LADealII::VectorType &y,
+    LADealII::VectorType &y_dot,
+    LADealII::VectorType &,
+    LADealII::VectorType &,
+    bool adaptive_refinement)
+{
+  SolutionTransfer<dim, LADealII::VectorType, DoFHandler<dim,spacedim> > sol_tr(*dof_handler);
```

Comment:
Can we use only one solution transfer? It allows you to call it with a vector of solutions...